haystack/components/eval/sas_evaluator.py

```diff
@@ -0,0 +1,156 @@
+from typing import Any, Dict, List, Optional
+
+from numpy import mean as np_mean
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice, expit
+from haystack.utils.auth import Secret, deserialize_secrets_inplace
+
+from .preprocess import _preprocess_text
+
+with LazyImport(message="Run 'pip install scikit-learn \"sentence-transformers>=2.2.0\"'") as metrics_import:
+    from sentence_transformers import CrossEncoder, SentenceTransformer, util
+    from transformers import AutoConfig
+
+
+@component
+class SASEvaluator:
+    """
+    SASEvaluator computes the Semantic Answer Similarity (SAS) between a list of predictions and a list of labels.
+    It's usually used in Retrieval Augmented Generation (RAG) pipelines to evaluate the quality of the generated answers.
+
+    The SAS is computed using a pre-trained model from the Hugging Face model hub. The model can be either a
+    Bi-Encoder or a Cross-Encoder. The choice of the model is based on the `model` parameter.
+    The default model is `sentence-transformers/paraphrase-multilingual-mpnet-base-v2`.
+    """
+
+    def __init__(
+        self,
+        labels: List[str],
+        model: str = "sentence-transformers/paraphrase-multilingual-mpnet-base-v2",
+        regexes_to_ignore: Optional[List[str]] = None,
+        ignore_case: bool = False,
+        ignore_punctuation: bool = False,
+        ignore_numbers: bool = False,
```

Comment:
I feel it would be a better idea to move this preprocessing into its own component? The transformations are not specific to this component as the statistical evaluator also uses it. Separating it will also let us add other transformations without breaking this component.